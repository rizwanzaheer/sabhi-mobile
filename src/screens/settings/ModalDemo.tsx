import * as React from 'react'
import { Container, Modal, Text, Constants } from '@kancha/kancha-ui'
import { NavigationScreenProps } from 'react-navigation'
import { Colors } from '../../theme'

interface Props extends NavigationScreenProps {}

const ModalDemoScreen: React.FC<Props> = ({ navigation }) => {
  return (
    <Modal dismiss={() => navigation.goBack()} scrollEnabled>
      <Container padding>
        <Container marginBottom>
          <Text type={Constants.TextTypes.H3}>Modal Demo</Text>
        </Container>
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
        <Container
          h={80}
          backgroundColor={Colors.LIGHTEST_GREY}
          br={10}
          marginBottom={10}
        />
      </Container>
    </Modal>
  )
}

export default ModalDemoScreen
