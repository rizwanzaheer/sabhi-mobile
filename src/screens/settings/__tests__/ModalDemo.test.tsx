import 'react-native'
import React from 'react'
import ModalDemo from '../../settings/ModalDemo'
import { render } from 'react-native-testing-library'

const navigation = {
  goBack: jest.fn(),
}

it('renders correctly', () => {
  // @ts-ignore
  const tree = render(<ModalDemo navigation={navigation} />).toJSON()
  expect(tree).toMatchSnapshot()
})
