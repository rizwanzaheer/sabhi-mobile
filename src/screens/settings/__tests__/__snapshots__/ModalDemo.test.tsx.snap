// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "alignSelf": "stretch",
        "backgroundColor": "#EFEEF3",
        "borderRadius": undefined,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 2,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 50,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "shadowColor": "#000000",
        "shadowOpacity": 0.2,
        "shadowRadius": 20,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": "center",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": 8,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#BBBBBB",
            "borderRadius": 5,
            "flex": undefined,
            "flexDirection": undefined,
            "height": 5,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": 50,
          }
        }
      />
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": undefined,
          "zIndex": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 8,
            "maxWidth": 300,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "height": undefined,
              "justifyContent": "center",
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": 10,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#BBBBBB",
                    "fontSize": 40,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ÔêÜ
            </Text>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": "center",
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 18,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView>
      <View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": 15,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 15,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#333333",
                  "flexWrap": "wrap",
                  "fontSize": 20,
                }
              }
            >
              Modal Demo
            </Text>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#EFEEF3",
                "borderRadius": 10,
                "flex": undefined,
                "flexDirection": undefined,
                "height": 80,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 10,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
